_posts/2017-06-26-windows-containers-your-way.md

```diff
@@ -0,0 +1,111 @@
+
+![MSBuild. Containers your way](/img/windows-containers-your-way/msbuild_containers_your_way.jpg)
+
+_View over Seattle port. Containers for everybody_ 
+
+## Container crash course
+
+Docker Containers are sort of new thing for the .NET developers. My first run in with them was when Microsoft demoed a .NET app running inside a Windows container during Build 2015. Therefore, like many other .NET developers, I am just beginning my Docker journey.
+
+Short recap: Containers are the next evolutionary step in virtualization technology. While the current virtualization technology enables IT departments to run multiple operating systems inside another operating system on top of single hardware, containers _share_ the host operating system kernel. This means that we do not have to duplicate the guest O/S in containers as we do today with traditional virtual machines. A container does not include a full O/S, but instead only the settings and libraries that are needed for the app the run. This is illustrated in the image below:
+
+<div align="center">
+    <img src="/img/windows-containers-your-way/vm_vs_container.png"/>
+</div>
+
+The virtualization technology from the 2000s was revolutionary as it made possible to run more apps on the same hardware. Containers take this trend further, as containers can be deployed even more densely as the containers have a much smaller footprint in terms of size when compared to a VM image. 
+
+However, the virtualization technology from the 2000s did not affect developers directly. Containers do as they change both how software is developed and deployed.
+
+## Driving factors for Windows Containers, developer perspective
+
+So why should a .NET developer care about containers? Here are a few reasons:
+* **Microservice architecture.** When developing to an actual microservice architecture (opposed to something that's just renamed Service Oriented Architecture) Docker containers help to manage the development environment when there are multiple microservices and each of which have their own database. This can be done without containers too, but they do ease up the task.
+* **"Works on my machine"** is something that we have all heard of and well... said. Containers ease up the pain as the development time testing can be done within exact same kind of environment as the production environment. This is because containers contain not only the app but also the environment that the app runs in.
```

Comment:
"is something that we have all heard of and well... said" minusta vähän vaikeasti luettava lause, "well said":llä on oma tarkoituksensa kuitenkin. Miten olisi "is something that we have all heard... and said"?